File: external/ble_wificonfig/Server/src/WifiInfo.cpp
#include "WifiInfo.h"

std::string WifiInfo::getBssid() {
    return this->m_bssid;
}
    
void WifiInfo::setBssid(std::string bssid) {
    this->m_bssid = bssid;
}
    
std::string WifiInfo::getFrequency() {
    return this->m_frequency;
}
   
void WifiInfo::setFrequency(std::string frequency) {
    this->m_frequency = frequency;
}
    
std::string WifiInfo::getSignalLevel() {
    return this->m_signalLevel;
}
    
void WifiInfo::setSignalLevel(std::string signalLevel) {
    this->m_signalLevel = signalLevel;
}
    
std::string WifiInfo::getFlags() {
    return this->m_flags;
}
    
void WifiInfo::setFlags(std::string flags) {
    this->m_flags = flags;
}
    
std::string WifiInfo::getSsid() {
    return this->m_ssid;
}
    
void WifiInfo::setSsid(std::string ssid) {
    this->m_ssid = ssid;
}

void WifiInfo::addJsonToRoot(rapidjson::Document &document,rapidjson::Value &root) {
    rapidjson::Value object(rapidjson::kObjectType);

    //object.AddMember("bssid",rapidjson::StringRef(m_bssid.c_str()),document.GetAllocator());
    //object.AddMember("frequency", rapidjson::StringRef(m_frequency.c_str()), document.GetAllocator());
    object.AddMember("level", rapidjson::StringRef(m_signalLevel.c_str()), document.GetAllocator());
    object.AddMember("flags", rapidjson::StringRef(m_flags.c_str()), document.GetAllocator());
    object.AddMember("id", rapidjson::StringRef(m_ssid.c_str()), document.GetAllocator());

    root.PushBack(object, document.GetAllocator());
}

File: external/app/src/RockHome/app/src/main/java/com/rockchip/smart/rockhome/WifiInfo.java
package com.rockchip.smart.rockhome;

public class WifiInfo implements Comparable<WifiInfo> {
    public static final String BSSID = "bssid";
    public static final String FREQUENCY = "frequency";
    public static final String SIGNALLEVEL = "signalLevel";
    public static final String FLAGS = "flags";
    public static final String SSID = "ssid";

    private boolean mIsConnecting;
    private boolean mIsConnected;

    private String bssid;
    private String frequency;
    private String signalLevel;
    private String flags;
    private String ssid;

    public String getBssid() {
        return bssid;
    }
    public void setBssid(String bssid) {
        this.bssid = bssid;
    }
    public String getFrequency() {
        return frequency;
    }
    public void setFrequency(String frequency) {
        this.frequency = frequency;
    }
    public String getSignalLevel() {
        return signalLevel;
    }
    public void setSignalLevel(String signalLevel) {
        this.signalLevel = signalLevel;
    }
    public String getFlags() {
        return flags;
    }
    public void setFlags(String flags) {
        this.flags = flags;
    }
    public String getSsid() {
        return ssid;
    }
    public void setSsid(String ssid) {
        this.ssid = ssid;
    }

    public boolean isConnecting() {
        return mIsConnecting;
    }

    public void setConnecting(boolean isConnecting) {
        this.mIsConnecting = isConnecting;
    }

    public boolean isConnected() {
        return mIsConnected;
    }

    public void setConnected(boolean isConnected) {
        this.mIsConnected = isConnected;
    }

    @Override
    public String toString() {
        StringBuilder builder = new StringBuilder();
        builder.append("{\"bssid\":\"").append(bssid).append(",\"frequency\":").append(frequency)
                .append(",\"signalLevel\":").append(signalLevel)
                .append(",\"flags\":").append(flags)
                .append(",\"ssid\":").append(ssid)
                .append("}");
        return builder.toString();
    }

    @Override
    public int compareTo(WifiInfo wifiInfo) {
        try {
            int level = Integer.parseInt(signalLevel);
            int compLevel = Integer.parseInt(wifiInfo.getSignalLevel());
            return level == compLevel ? 0 : (level > compLevel ? -1 : 1);
        } catch (Exception e) {

        }
        return 0;
    }
}
